compiler/compile/OMRSymbolReferenceTable.hpp

```diff
@@ -672,6 +672,30 @@ class SymbolReferenceTable
     */
    TR::SymbolReference * findOrFabricateShadowSymbol(TR_OpaqueClassBlock *containingClass, TR::DataType type, uint32_t offset, bool isVolatile, bool isPrivate, bool isFinal, const char * name, const char * signature);
 
+    /** \brief
+    *     Returns an array shadow symbol reference fabricated for the field of a flattened array element.
+    *
+    *  \param arrayComponentClass
+    *     The array component class that contains the field.
+    *  \param type
+    *     The data type of the field.
+    *  \param fieldOffset
+    *     The offset of the field.
+    *  \param isVolatile
+    *     Specifies whether the field is volatile.
+    *  \param isPrivate
+    *     Specifies whether the field is private.
+    *  \param isFinal
+    *     Specifies whether the field is final.
+    *  \param fieldName
+    *     The name of the field.
+    *  \param fieldSignature
+    *     The signature of the field.
+    *  \return
+    *     Returns an array shadow symbol reference fabricated for the field of a flattened array element.
+    */
+   TR::SymbolReference * findOrFabricateFlattenedArrayShadowSymbol(TR_OpaqueClassBlock *arrayComponentClass, TR::DataType type, uint32_t fieldOffset, bool isVolatile, bool isPrivate, bool isFinal,  const char *fieldName, const char *fieldSignature);
```

Comment:
Sorry for the post-review comment.  Thinking about the name again, I wonder whether a fully descriptive name like `findOrFabricateFlattenedArrayElementFieldShadowSymbol` would be better - or maybe `findOrFabricateFieldOfFlattenedElementShadowSymbol` - as it's not a shadow symbol for the array as a whole that's returned, but a shadow symbol for a field of an array element.

Reply:
Addressed in 2ed16c8. The API is also updated to remove `isFinal` and `isVolatile` based on the review comments in https://github.com/eclipse-openj9/openj9/pull/13977.